projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/WPCOM_REST_API_V2_Endpoint_JITM_V2_Test.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+require_once dirname( __DIR__, 2 ) . '/lib/Jetpack_REST_TestCase.php';
+
+/**
+ * Tests for JITM V2 REST API Endpoints.
+ *
+ * @package automattic/jetpack
+ */
+class WPCOM_REST_API_V2_Endpoint_JITM_V2_Test extends Jetpack_REST_TestCase {
+
+	/**
+	 * Mock user ID.
+	 *
+	 * @var int
+	 */
+	private static $user_id = 0;
+
+	/**
+	 * Store the Patchwork redefine handle
+	 *
+	 * @var mixed
+	 */
+	private $patchwork_handle;
+
+	/**
+	 * Create shared database fixtures.
+	 *
+	 * @param WP_UnitTest_Factory $factory Fixture factory.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		static::$user_id = $factory->user->create( array( 'role' => 'administrator' ) );
+	}
+
+	/**
+	 * Setup the environment for a test.
+	 */
+	public function set_up() {
+		parent::set_up();
+		wp_set_current_user( static::$user_id );
+
+		// Mock current_user_can to return true for install_plugins when testing with WPCOMSH.
+		// This is required because the Pre_Connection_JITM->get_messages() method
+		// uses `current_user_can` to check if the user has the `install_plugins` capability.
```

Comment:
I'm trying to figure out _why_ we have this cap check. Tests assert on it but don't provide any context

https://github.com/Automattic/jetpack/blob/trunk/projects/packages/jitm/tests/php/Pre_Connection_JITM_Test.php#L67-L70

Reply:
https://github.com/Automattic/jetpack/pull/42429